### PR TITLE
Parser: Allow trailing `,` in all lists.

### PR DESCRIPTION
**Pull request by gzanitti**
Solves #1853

### CODE REVIEW

**Review by chriseth — APPROVED** [review]
Looks good! If this is all lists where commas are useful, then please change the main comment from "relates to" to "solves" or "closes". Then the issue will be automatically closed when the PR is merged.

**Comment by chriseth** [comment]
Why does it not run the tests? Maybe because you are not using a branch on the powdr-labs repository but in your own fork?

**Comment by gzanitti** [comment]
> Why does it not run the tests? Maybe because you are not using a branch on the powdr-labs repository but in your own fork?

No idea, never happened before. But I had to solve a conflict and they are running now.

File: parser/src/lib.rs [hunk 1]
```diff
@@ -551,6 +551,86 @@ namespace N(8);
...
+"#;
+        let expected = r#"
+machine Main(a: Byte, b: Byte) {
+    reg pc[@pc];
+    reg X[<=];
+    function get a, step -> b {
+    A <== mload(a, step);
+    return A;
+    }
+    instr assert_eq X, Y{ X = Y }
```
Comment (chriseth) [inline parser/src/lib.rs:627]: This doesn't look right, was it like that before?

**Review by chriseth — APPROVED** [review]

**Pull request merged**